web_hierarchy_list/static/src/hierarchy_list_x2many_field.esm.js

```diff
@@ -0,0 +1,78 @@
+import {X2ManyField, x2ManyField} from "@web/views/fields/x2many/x2many_field";
+import {HierarchyListRenderer} from "./hierarchy_list_renderer.esm";
+import {RelationalModel} from "@web/model/relational_model/relational_model";
+import {registry} from "@web/core/registry";
+import {treatHierarchyListArch} from "./hierarchy_list_arch_utils.esm";
+import {useService} from "@web/core/utils/hooks";
+import {useState} from "@odoo/owl";
+
+export class HierarchyListX2manyField extends X2ManyField {
+    static components = {
+        ...X2ManyField.components,
+        HierarchyListRenderer,
+    };
+    static template = "web_hierarchy_list.X2ManyField";
+
+    setup() {
+        super.setup();
+        treatHierarchyListArch(
+            this.archInfo,
+            this.props.record.fields[this.props.name].relation,
+            this.archInfo.fields
+        );
+        this.state = useState({
+            parent: false,
+        });
+        const services = {};
+        for (const key of RelationalModel.services) {
+            services[key] = useService(key);
```

Comment:
i wish that we could just get the services directly from the model itself. Technically, it could work, because the new RelationalModel is done in the setup, so we can actually call useService in the model constructor, but it feels wrong, which is why we kind of have to do this.  still feels awkward